src/components/GlobalSearch/GlobalSearchComponent.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useContext, useCallback, useState } from 'react';
+import { ClientSDKContext } from '../../context/clientSDKContext';
+import { globalSearch, SearchResource, SearchForFilter } from './globalSearch';
+import { Search } from '../common/Search/Search';
+import { PureObject } from '../..';
+import _ from 'lodash';
+import styled from 'styled-components';
+import { Icon } from 'antd';
+import { withDefaultTheme } from '../../hoc';
+
+type GlobalSearchStrings = {
+  assets: string,
+  timeseries: string,
+  events: string,
+  files: string,
+  noDescription: string
+};
+
+const defaultStrings: GlobalSearchStrings = {
+  assets: 'Assets',
+  timeseries: 'Timeseries',
+  events: 'Events',
+  files: 'Files',
+  noDescription: "[No Description]",
+};
+
+type RenderSearchItem = (strings: GlobalSearchStrings, item: SearchResource) => React.ReactNode;
+
+type GlobalSearchProps = {
+  onError?: (error: any) => void;
+  searchForFilter?: SearchForFilter;
+  strings?: Partial<GlobalSearchStrings>;
+  renderSearchResult?: (
+    strings: GlobalSearchStrings,
+    searchForFilter: SearchForFilter,
+    items: SearchResource[],
+    renderSearchItem: RenderSearchItem
+  ) => React.ReactNode;
+  renderSearchItem?: RenderSearchItem;
+};
+
+function titleFromResource(strings: GlobalSearchStrings, resource: SearchResource) {
+  switch(resource.resourceType) {
+    case "asset": return resource.name;
+    case "timeSeries": return resource.name;
```

Comment:
you know it can be done it declarative way, right?
```
return {
asset: name,
timeseries: name
}[resourceType]

```

Reply:
Typescript would not know about the discriminator in that case, so that `resource.name` or `resource.description` would fail to compile, unless you casted it to any first, but sure you could do that. However I think the intent is more obvious with a switch statement also.